sdk/data/azappconfig/internal/auth/policy_hmac_auth.go

```diff
@@ -100,47 +99,64 @@ func getHMAC(content string, key []byte) (string, error) {
 }
 
 // ParseConnectionString parses the provided connection string.
-func ParseConnectionString(connectionString string) (endpoint string, credential string, secret []byte, err error) {
-	const connectionStringEndpointPrefix = "Endpoint="
-	const connectionStringCredentialPrefix = "Id="
-	const connectionStringSecretPrefix = "Secret="
-
-	var er error = errors.New("error parsing connection string")
-	var ept *string
-	var cred *string
-	var sec *[]byte
+// Returns endpoint, cred, secret or an error.
+func ParseConnectionString(connectionString string) (string, string, []byte, error) {
```

Comment:
At some point you've got to be considering returning this as a `struct`...

Reply:
I thought about it, but it seems like extra ceremony for something that's internal.